apps/pipelines/models.py

```diff
@@ -127,22 +163,54 @@ def _create_pipeline_run(self, input: PipelineState, session: ExperimentSession)
             session=session,
         )
 
-    def _save_message_to_history(self, session: ExperimentSession, message: str, type_: ChatMessageType):
-        ChatMessage.objects.create(
+    def _save_message_to_history(self, session: ExperimentSession, message: str, type_: ChatMessageType) -> ChatMessage:
+        return ChatMessage.objects.create(
             chat=session.chat,
             message_type=type_.value,
             content=message,
         )
         # TODO: Add tags here?
 
+    @transaction.atomic()
+    def create_new_version(self, *args, **kwargs):
+        version_number = self.version_number
+        self.version_number = version_number + 1
+        self.save()
+        pipeline_version = super().create_new_version(*args, **kwargs)
```

Comment:
you could pass in `save=False` where and then save the whole thing below instead of saving twice.

Reply:
Nice catch. I'll update

Reply:
See https://github.com/dimagi/open-chat-studio/pull/735/commits/4f9c29cea585326c02dd07aad49f110f20bf227c